src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -158,6 +181,795 @@ MultiLogicalPlanCreate(Query *originalQuery, Query *queryTree)
 
 
 /*
+ * MultiSubqueryPlanTree gets the query objects and returns logical plan
+ * for subqueries.
+ *
+ * We currently have two different code paths for creating logic plan for subqueries:
+ *   (i) subquery pushdown
+ *   (ii) single relation repartition subquery
+ *
+ * In order to do create the logical plan, we follow the algorithm below:
+ *    -  If subquery pushdown planner can plan the query
+ *        -  We're done, we create the multi plan tree and return
+ *    -  Else
+ *       - If the query is not eligible for single table repartition subquery planning
+ *            - Throw the error that the subquery pushdown planner generated
+ *       - If it is eligible for single table repartition subquery planning
+ *            - Check for the errors for single table repartition subquery planning
+ *                - If no errors found, we're done. Create the multi plan and return
+ *                - If found errors, throw it
```

Comment:
> Well, that would mean that we remove all the single repartition error messages. I think keeping it as it is might be better. Otherwise, someone who is trying to run single repartition subquery would always see subquery pushdown error messages.

The reverse is also correct. Someone who is trying to run subquery pushdown query with a single table will see errors related repartition subquery planning. My argument is that for our use cases, it is more probable that somebody will want to run a subquery pushdown query than a  single table repartition subquery.